tableaudocumentapi/workbook.py

```diff
@@ -95,12 +109,45 @@ def save_as(self, new_filename):
     # Private API.
     #
     ###########################################################################
-    def _prepare_datasources(self, xmlRoot):
+    @staticmethod
+    def _prepare_datasource_index(datasources):
+        retval = weakref.WeakValueDictionary()
+        for datasource in datasources:
+            retval[datasource.name] = datasource
+
+        return retval
+
+    @staticmethod
+    def _prepare_datasources(xml_root):
         datasources = []
 
         # loop through our datasources and append
-        for datasource in xmlRoot.find('datasources'):
+        for datasource in xml_root.find('datasources'):
             ds = Datasource(datasource)
             datasources.append(ds)
 
         return datasources
+
+    @staticmethod
+    def _prepare_worksheets(xml_root, ds_index):
+        worksheets = []
+        worksheets_element = xml_root.find('.//worksheets')
+        if worksheets_element is None:
+            return worksheets
+
+        for worksheet_element in worksheets_element:
+            worksheet_name = worksheet_element.attrib['name']
+            worksheets.append(worksheet_name)  # TODO: A real worksheet object, for now, only name
+
+            dependencies = worksheet_element.findall('.//datasource-dependencies')
+            if dependencies is None:
```

Comment:
This check will never succeed, findall returns an empty list if no matches

The reason it works is because an iteration over an empty list will silently just skip.

```
dependancies = worksheet_element.findall('.//datasource-dependencies')
if not dependencies:
    continue
```